app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -64,6 +64,7 @@ def load_resource_from_solr
 
     # routed to /files/new
     def new
+      set_variables_for_new_form
```

Comment:
The @user_collections which comes from hydra-collections returns only the list of collections the user has permissions controlled by line before_action on [line 43](https://github.com/projecthydra/sufia/blob/upload_with_collection/app/controllers/concerns/sufia/files_controller_behavior.rb) (find_collections_with_edit_access).  Then in set_variables_for_new_form, it checks that the passed in collection_id is in the @user_collections list.  Seems sufficient to me.  Does that work for you?